dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AdminPrivilegesFilter.java

```diff
@@ -0,0 +1,36 @@
+package io.dockstore.webservice.resources;
+
+import java.io.IOException;
+import java.security.Principal;
+
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerResponseContext;
+import javax.ws.rs.container.ContainerResponseFilter;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.UriInfo;
+import javax.ws.rs.ext.Provider;
+
+import io.dockstore.webservice.core.User;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Provider
+public class AdminPrivilegesFilter implements ContainerResponseFilter {
+    private static final Logger LOG = LoggerFactory.getLogger(AdminPrivilegesFilter.class);
+    @Context
+    UriInfo uriInfo;
+
+    @Override
+    public void filter(ContainerRequestContext requestContext, ContainerResponseContext responseContext) throws IOException {
+        if (requestContext != null) {
+            Principal principal = requestContext.getSecurityContext().getUserPrincipal();
+            if (principal != null) {
```

Comment:
You could change this to `if (principal instanceof User) {`. If present, I'm guessing the principal must always be a User, but even so, it would give you more confidence for the next line where you cast it.